File: fix_jsons.py
import json
import glob
import re
import os

assets_dir = 'c:/la-previa/assets'

# Find all events json
event_files = glob.glob(f'{assets_dir}/events*.json')

for filepath in event_files:
    with open(filepath, 'r', encoding='utf-8') as f:
        data = json.load(f)
    
    modified = False
    for event in data.get('events', []):
        # 1. Remove EVENTO: / EVENT: from templates
        if 'template' in event:
            new_template = re.sub(r'^(EVENTO:\s*|EVENT:\s*)', '', event['template'])
            if new_template != event['template']:
                event['template'] = new_template
                modified = True
                
        # 2. Fix multiplier if title is Tragos Dobles / Double Drinks
        title = event.get('title', '')
        if 'Tragos Dobles' in title or 'Double Drinks' in title:
            if 'variables' in event and 'MULTIPLIER' in event['variables']:
                if event['variables']['MULTIPLIER'] != ['2']:
                    event['variables']['MULTIPLIER'] = ['2']
                    modified = True

    if modified:
        with open(filepath, 'w', encoding='utf-8') as f:
            json.dump(data, f, ensure_ascii=False, indent=2)

# Fix English words in Spanish question jsons (shots -> chupitos)
question_files = [f for f in glob.glob(f'{assets_dir}/questions*.json') if '_en' not in f]
for filepath in question_files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Simple regex to replace 'shots' with 'chupitos' (case insensitive)
    if re.search(r'\b(shots|shot)\b', content, re.IGNORECASE):
        # Be careful not to replace ID or English keys if any, but since it's ES, should be safe in text. 
        # But wait, "shots" might be perfectly matched in text
        data = json.loads(content)
        modified = False
        for q in data.get('questions', []):
            if 'text' in q:
                new_text = re.sub(r'\bshots\b', 'chupitos', q['text'], flags=re.IGNORECASE)
                new_text = re.sub(r'\bshot\b', 'chupito', new_text, flags=re.IGNORECASE)
                if new_text != q['text']:
                    q['text'] = new_text
                    modified = True
            
        if modified:
            with open(filepath, 'w', encoding='utf-8') as f:
                json.dump(data, f, ensure_ascii=False, indent=2)

print("Finished fixing events and question jsons")

File: fix_jsons.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final assetsDir = Directory('c:/la-previa/assets');
  final entities = assetsDir.listSync(recursive: false);

  for (var entity in entities) {
    if (entity is File && entity.path.endsWith('.json')) {
      final fileName = entity.path.split(Platform.pathSeparator).last;
      bool isEvent = fileName.startsWith('events');
      bool isQuestion = fileName.startsWith('questions') && !fileName.contains('_en');
      bool modified = false;
      
      final content = await entity.readAsString();
      Map<String, dynamic> data;
      try {
        data = jsonDecode(content);
      } catch (e) {
        continue;
      }

      if (isEvent && data.containsKey('events')) {
        final events = data['events'] as List<dynamic>;
        for (var event in events) {
          if (event is Map && event.containsKey('template')) {
            String template = event['template'];
            String newTemplate = template.replaceFirst(RegExp(r'^(EVENTO:\s*|EVENT:\s*)'), '');
            if (newTemplate != template) {
              event['template'] = newTemplate;
              modified = true;
            }
          }
          if (event is Map) {
            String title = event['title'] ?? '';
            if (title.contains('Tragos Dobles') || title.contains('Double Drinks')) {
              if (event.containsKey('variables') && event['variables'].containsKey('MULTIPLIER')) {
                List<dynamic> multiplier = event['variables']['MULTIPLIER'];
                if (multiplier.length != 1 || multiplier[0] != "2") {
                  event['variables']['MULTIPLIER'] = ["2"];
                  modified = true;
                }
              }
            }
          }
        }
      }

      if (isQuestion && data.containsKey('questions')) {
        final questions = data['questions'] as List<dynamic>;
        for (var q in questions) {
          if (q is Map && q.containsKey('text')) {
            String text = q['text'];
            String newText = text.replaceAll(RegExp(r'\bshots\b', caseSensitive: false), 'chupitos');
            newText = newText.replaceAll(RegExp(r'\bshot\b', caseSensitive: false), 'chupito');
            if (newText != text) {
              q['text'] = newText;
              modified = true;
            }
          }
        }
      }

      if (modified) {
        // Formatter config isn't needed, standard encode with spaces
        final JsonEncoder encoder = new JsonEncoder.withIndent('  ');
        await entity.writeAsString(encoder.convert(data));
      }
    }
  }
}
